app/src/main/kotlin/com/wire/android/ui/home/userprofile/UserProfileScreen.kt

```diff
@@ -1,16 +1,343 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
-import com.wire.android.ui.common.UnderConstructionScreen
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.wire.android.R
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.common.Icon
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.common.UserStatus
+import com.wire.android.ui.common.UserStatusDot
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryButton
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.selectableBackground
+import com.wire.android.ui.common.textfield.WirePrimaryButton
+import com.wire.android.ui.theme.wireTypography
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun UserProfileScreen() {
-    UnderConstructionScreen(screenName = "UserProfileScreen")
+fun UserProfileScreen(viewModel: UserProfileViewModel) {
+    val uiState by viewModel.state.collectAsState()
+
+    with(uiState) {
+        renderUserProfileScreen(state = this, viewModel = viewModel)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun renderUserProfileScreen(state: UserProfileState, viewModel: UserProfileViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
+            .background(MaterialTheme.colorScheme.background)
+    ) {
+
+        renderTopBar(viewModel)
+        renderHeader(state, viewModel)
+        renderStatusesRow(state)
+        renderOtherAccountsList(state, viewModel)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ColumnScope.renderTopBar(viewModel: UserProfileViewModel) {
+    CenterAlignedTopAppBar(
+        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.background,
+            titleContentColor = MaterialTheme.colorScheme.onBackground,
+            actionIconContentColor = MaterialTheme.colorScheme.onBackground,
+            navigationIconContentColor = MaterialTheme.colorScheme.onBackground
+        ),
+        navigationIcon = {
+            IconButton(onClick = { viewModel.close() }) {
+                Icon(
+                    imageVector = Icons.Filled.Close,
+                    contentDescription = stringResource(R.string.user_profile_close_description),
+                    tint = MaterialTheme.colorScheme.onBackground
```

Comment:
Tint isn't necessary because `CenterAlignedTopAppBar` has `colors` parameter and handles content colors itself